### PR TITLE
updates to modeling and asdf docs

### PR DESCRIPTION
**Pull request by cshanahan1**
Made some small changes to modeling docs. I also added a section in the ASDF docs of how to write a schema and a tag for a new object, using a model as an example.

### CODE REVIEW

File: docs/io/misc.rst [hunk 1]
```diff
@@ -222,7 +222,100 @@ Not all ``astropy`` types are currently serializable by ASDF. Attempting to
...
+All schemas for transforms are currently defined within the ASDF standard.
+Any new serializable transforms must have a corresponding new
+schema here. Let's consider a new model called ``MyModel``, a new model in
+``astropy.modeling.functional_models`` that has two parameters ``amplitude``
+and ``x_0``. We would like to strictly require both of these parameters be set.
+We would also like to specify that these parameters can either be numeric type,
+or ``astropy.units.quantity`` type. A schema describing this
+model would look like::
+
+        %YAML 1.1
```
Comment (pllim) [inline docs/io/misc.rst:244]: Use 4 spaces?

File: docs/io/misc.rst [hunk 2]
```diff
@@ -222,7 +222,100 @@ Not all ``astropy`` types are currently serializable by ASDF. Attempting to
...
+new custom types.
+
+The next component is the tag class. This class must have a ``to_tree`` method
+in which the required attributes of the object in question are obtained, and a
+``from_tree`` method which reconstructs the object based on the parameters
+written to the ASDF file. ``astropy`` Models inherit from the
+``TransformType`` base class tag, which takes care of attributes (e.g ``name``,
+``bounding_box``, ``n_inputs``) that are common to all or many Model classes to
+limit redundancy in individual tags. Each individual model tag then only has
+to obtain and set model-specific parameters.::
```
Comment (pllim) [inline docs/io/misc.rst:288]:
```suggestion
to obtain and set model-specific parameters::
```

File: docs/modeling/models.rst [hunk 3]
```diff
@@ -557,3 +558,6 @@ To read the file and create the model:
         angle
         -----
          23.7
+Compound models can also be serialized. Please note that some model attributes (e.g ``meta``,
```
Comment (pllim) [inline docs/modeling/models.rst:561]:
```suggestion

Compound models can also be serialized. Please note that some model attributes (e.g ``meta``, 
```

**Comment by nden** [comment]
LGTM. Since it's a documentation PR is it possible it goes in 4.1?

**Comment by pllim** [comment]
Branching has happened, so I'll leave that to @bsipocz 's discretion.

**Comment by nden** [comment]
There's no urgent need for this to go in 4.1, I just remember from past releases doc updates are OK after branching.

**Comment by lglattly** [comment]
@pllim and @cshanahan1 unfortunately my copy editing work for Astropy is on pause right now while I put together a proposal for the finance committee to continue work on the new grant -- I just wanted to let you know that here so I don't hold up this PR!

**Comment by pllim** [comment]
Thanks for letting us know, @lglattly . Good luck!

**Comment by astrofrog** [comment]
I'll backport this to 4.1 since approved

**Pull request merged**